packages/ckeditor5-dev-release-tools/tests/utils/publishpackageonnpmcallback.js

```diff
@@ -118,5 +118,24 @@ describe( 'dev-release-tools/utils', () => {
 					}
 				);
 		} );
+
+		it( 'should not remove a package directory and not throw error when publishing on npm failed with code 409', () => {
+			stubs.devUtils.tools.shExec.rejects( new Error( 'code E409' ) );
+
+			const packagePath = '/workspace/ckeditor5/packages/ckeditor5-foo';
+
+			return publishPackageOnNpmCallback( packagePath, { npmTag: 'nightly' } )
+				.then(
+					() => {
+						// We expect publishPackageOnNpmCallback to not throw, so this error should be thrown instead.
+						throw new Error( 'Expected to be rejected.' );
+					}
+				)
+				.catch( error => {
+					expect( stubs.fs.remove.callCount ).to.equal( 0 );
+					expect( error ).to.be.an( 'Error' );
+					expect( error.message ).to.equal( 'Expected to be rejected.' );
+				} );
+		} );
```

Comment:
```suggestion
			await publishPackageOnNpmCallback( packagePath, { npmTag: 'nightly' } )

			expect( stubs.fs.remove.callCount ).to.equal( 0 );
		} );
```

Couldn't it be simplified?

Reply:
It could. I just followed the structure of tests that we use in the project.